src/main/java/org/opensearch/security/support/JsonFlattener.java

```diff
@@ -0,0 +1,55 @@
+package org.opensearch.security.support;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.opensearch.core.common.Strings;
+import org.opensearch.security.DefaultObjectMapper;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class JsonFlattener {
+
+    public static Map<String, Object> flattenAsMap(String jsonString) {
+        try {
+            final byte[] bytes = jsonString.getBytes("utf-8");
+            final TypeReference<Map<String, Object>> typeReference = new TypeReference<>() {
+            };
+            final Map<String, Object> jsonMap = DefaultObjectMapper.objectMapper.readValue(bytes, typeReference);
+            final Map<String, Object> flattenMap = new LinkedHashMap<>();
```

Comment:
`DefaultObjectMapper.objectMapper.readValue` supports strings no need to take bytes Jackson does the same.